tracers: fix err in 4byte, add unigram/bigram/trigram opcode analysis…

yoomee1313 opened the pull request:
## Proposed changes
This PR is derived from https://github.com/ethereum/go-ethereum/pull/16954.
- It fixes an error in the `4byteTracer`.
- It adds `unigram_tracer.js` which just counts opcodes.
- It adds `bigram_tracer.js` which counts bigrams.
- It adds `trigram_tracer.js` which counts trigrams.

New Tracers Output
* unigram_tracer.js
![image](https://user-images.githubusercontent.com/64360042/99624402-1b768800-29e3-11eb-94c1-45eedd29d410.png)
* bigram_tracer.js
![image](https://user-images.githubusercontent.com/64360042/99624440-292c0d80-29e3-11eb-8951-c50c61db6a26.png)
* trigram_tracer.js
![image](https://user-images.githubusercontent.com/64360042/99624476-35b06600-29e3-11eb-98db-8c7ee65750dd.png)

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

KimKyungup commented on node/cn/tracers/internal/tracers/bigram_tracer.js:
```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 The go-ethereum Authors
```
Please change license descriptions.
You can refer other file license.

yoomee1313 replied:
I change license descriptions of tracer javascript files. Please check those changes @KimKyungup

KimKyungup commented:
@yoomee1313 Please check and show the result of new tracers.

yoomee1313 commented:
@KlaytnDev I added the results of new tracers. Also, I changed descriptions of tracers in this commit (https://github.com/klaytn/klaytn/pull/752/commits/98a1ae1f76da15ca5ca92d5e0f5ee1642f671d48).

jeongkyun-oh commented on node/cn/tracers/internal/tracers/bigram_tracer.js:
```diff
@@ -0,0 +1,62 @@
+// Modifications Copyright 2020 The klaytn Authors
```
I think the license should be updated when we modified the file. If the file is exactly the same as the one in ethereum, we didn't put the modification comments.

yoomee1313 replied:
You're right.. I added klaytn copyright only to the modified files https://github.com/klaytn/klaytn/pull/752/commits/8a8c113dfca481a39163e4dc858039b9f42da3f4, Thanks!

kjhman21 commented:
@yoomee1313 @KimKyungup Will this new tracers be introduced in Klaytn Docs, right? 😄

yoomee1313 commented:
@kjhman21 Yes, updating Klaytn Docs(https://docs.klaytn.com/bapp/json-rpc/api-references/debug/tracing#debug_tracetransaction) is necessary when those new tracers are included in release

ehnuje commented:
> @kjhman21 Yes, updating Klaytn Docs(https://docs.klaytn.com/bapp/json-rpc/api-references/debug/tracing#debug_tracetransaction) is necessary when those new tracers are included in release

@yoomee1313 Not to miss updating the docs, how about making a JIRA item to track the task?

yoomee1313 commented:
> > @kjhman21 Yes, updating Klaytn Docs(https://docs.klaytn.com/bapp/json-rpc/api-references/debug/tracing#debug_tracetransaction) is necessary when those new tracers are included in release
> 
> @yoomee1313 Not to miss updating the docs, how about making a JIRA item to track the task?

@ehnuje Instead of making a JIRA item, i will tag a label to the PRs that need docs!

KimKyungup commented:
@yoomee1313 Please update klaytn-docs before merging this.
